### PR TITLE
The debug command line argument is now passed correctly in Haplotype caller -- fixed issue #4943

### PR DESCRIPTION
**Pull request by MartonKN**
The debug variable of AsseemblyResultSet wasn't set anywhere, and therefore the command line argument debug didn't propagate to the function buildEventMapsForHaplotypes in EventMap.java. I added the function setDebug to AssemblyResultSet, and set its value in HaplotypeCallerEngine.

### CODE REVIEW

**Comment by davidbenjamin** [comment]
@MartonKN don't forget to put in: closes #4943 for github.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyResultSet.java [hunk 1]
```diff
@@ -518,4 +518,9 @@ private void updateReferenceHaplotype(final Haplotype newHaplotype) {
         }
         return variationEvents;
     }
+
+    // Set the debug variable
```
[the remark this answers is not in the document]
Comment (MartonKN) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyResultSet.java]: Ok, changed it, thanks

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java [hunk 2]
```diff
@@ -536,7 +536,7 @@ public ActivityProfileState isActive( final AlignmentContext context, final Refe
 
         // run the local assembler, getting back a collection of information on how we should proceed
         final AssemblyResultSet untrimmedAssemblyResult =  AssemblyBasedCallerUtils.assembleReads(region, givenAlleles, hcArgs, readsHeader, samplesList, logger, referenceReader, assemblyEngine, aligner);
-
+        untrimmedAssemblyResult.setDebug(hcArgs.debug);
```
[the remark this answers is not in the document]
Comment (MartonKN) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java]: All set.

**Comment by MartonKN** [comment]
Closes #4943 for github

**Comment by MartonKN** [comment]
Finished the requests of @davidbenjamin

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java [hunk 3]
```diff
@@ -193,6 +193,7 @@ public void writeHeader(final VariantContextWriter vcfWriter, final Set<VCFHeade
 
         final AssemblyRegion assemblyActiveRegion = AssemblyBasedCallerUtils.assemblyRegionWithWellMappedReads(originalAssemblyRegion, READ_QUALITY_FILTER_THRESHOLD, header);
         final AssemblyResultSet untrimmedAssemblyResult = AssemblyBasedCallerUtils.assembleReads(assemblyActiveRegion, givenAlleles, MTAC, header, samplesList, logger, referenceReader, assemblyEngine, aligner);
+        untrimmedAssemblyResult.setDebug(MTAC.debug);
```
[the remark this answers is not in the document]
Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java]: @MartonKN This comment still needs to be addressed.

[the remark this answers is not in the document]
Comment (MartonKN) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java]: 👍 All set, please, check.

**Comment by davidbenjamin** [comment]
@MartonKN Looks good!

**Pull request merged**